### PR TITLE
fluent-bit: update to 3.0.4

### PR DESCRIPTION
**Pull request by herbygillot**
###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by pmetzger** [comment]
@herbygillot Any idea why the CI failed on MacOS 11?

File: sysutils/fluent-bit/Portfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ PortGroup           legacysupport               1.1
 
 legacysupport.newest_darwin_requires_legacy     15
 
-github.setup        fluent fluent-bit 2.2.2 v
+github.setup        fluent fluent-bit 3.0.4 v
```
Comment (reneeotten) [inline sysutils/fluent-bit/Portfile:14]: @herbygillot @l2dy most recent version appears to be 3.0.6 - perhaps update to that and see if the build failure on macOS-11 disappears? If not, one should probably still merge this PR - possibly restricting older systems to the 2.x branch; no point in keeping this PR hanging in the queue.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: sysutils/fluent-bit/Portfile (hunk 1 above)
Comment (l2dy) [reply sysutils/fluent-bit/Portfile]: > If not, one should probably still merge this PR - possibly restricting older systems to the 2.x branch; no point in keeping this PR hanging in the queue.

We could also set `known_fail` on older OSes, if branching or subport is not desired.

**Comment by pmetzger** [comment]
We're no longer running the CI on '11 anyway...

**Pull request merged**